pkg/distributor/distributor.go

```diff
@@ -371,7 +376,7 @@ func (d *Distributor) sendSamplesErr(ctx context.Context, ingester *ring.Ingeste
 	}
 
 	err = instrument.TimeRequestHistogram(ctx, "Distributor.sendSamples", d.sendDuration, func(ctx context.Context) error {
-		_, err := c.Push(ctx, req)
+		_, err := c.(ingester_client.IngesterClient).Push(ctx, req)
```

Comment:
I think I would put this cast right after the value is obtained from the pool, so that if it does happen to fail it's clear what happened.